app/src/components/billing/BillingContentTabs/PaymentMethods/PaymentMethods.tsx

```diff
@@ -0,0 +1,114 @@
+  const utils = api.useContext();
+
+  const setDefaultPaymentMethodMutation = api.payments.setDefaultPaymentMethod.useMutation();
+  const deletePaymentMethodMutation = api.payments.deletePaymentMethod.useMutation();
+
+  const [setDefaultPaymentMethod, isDefaultChangeLoading] = useHandledAsyncCallback(
+    async (paymentMethodId: string) => {
+      if (!selectedProject) return;
+
+      await setDefaultPaymentMethodMutation.mutateAsync({
+        projectId: selectedProject.id,
+        paymentMethodId,
+      });
+
+      await utils.payments.getPaymentMethods.invalidate();
+    },
+    [selectedProject, utils, setDefaultPaymentMethodMutation],
+  );
+
+  const [deletePaymentMethod, isDeleteLoading] = useHandledAsyncCallback(
+    async (paymentMethodId: string) => {
+      if (!selectedProject) return;
+
+      const res = await deletePaymentMethodMutation.mutateAsync({
+        projectId: selectedProject.id,
+        paymentMethodId,
+      });
+
+      maybeReportError(res);
+
+      await utils.payments.getPaymentMethods.invalidate();
+    },
+    [selectedProject, utils, setDefaultPaymentMethodMutation],
+  );
+
+  return (
+    <VStack alignItems="flex-start" w="full" px={8} spacing={4} pb={8}>
+      <HStack spacing="4" align="stretch">
+        {paymentMethods &&
+          paymentMethods.data.length &&
+          paymentMethods.data.map((method) => (
+            <Card key={method.id} width="240px">
+              <CardBody pb={2}>
+                <HStack justify="space-between">
+                  {method.card?.brand === "visa" ? (
+                    <Icon as={FaCcVisa} color={"gray.500"} boxSize={16} />
+                  ) : method.card?.brand === "mastercard" ? (
+                    <Icon as={FaCcMastercard} color={"gray.500"} boxSize={16} />
+                  ) : (
+                    <Icon as={FaCreditCard} color={"gray.500"} boxSize={16} />
+                  )}
+                  <VStack alignItems="end">
+                    <Text fontSize="xl" as="b">
+                      ****{method.card?.last4}
+                    </Text>
+                    <Text fontSize="sm">
+                      Expires: {method.card?.exp_month}/{method.card?.exp_year}
+                    </Text>
+                  </VStack>
+                </HStack>
+                <HStack justify="space-between" m={0} p={0}>
+                  {method.id === paymentMethods.defaultPaymentMethodId ? (
+                    <Text fontSize="sm" color="gray.400" py={2}>
+                      Default
+                    </Text>
+                  ) : (
+                    <Button
+                      onClick={() => {
+                        if (!isDefaultChangeLoading) setDefaultPaymentMethod(method.id);
+                      }}
+                      fontSize="sm"
+                      px={1}
+                      variant="ghost"
+                      _hover={{ bgColor: "transperent" }}
+                    >
```

Comment:
Instead of adding custom <Spinner /> elements in loading buttons, you can use Chakra's built-in `isLoading` prop. That should also disable the button while it's loading.

Reply:
So convenient, I did not know it! Changed in all the places.